hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/MapLoaderTest.java

```diff
@@ -296,161 +308,137 @@ public Iterable loadAllKeys() {
+                        }
+                        if (keys.contains("2")) {
+                            val.put("2", "2");
+                        }
+                        if (keys.contains("3")) {
+                            val.put("3", "3");
+                        }
+                        return val;
+                    }
+
+                    @Override
+                    public Iterable<String> loadAllKeys() {
+                        List<String> keys = new ArrayList<String>();
+                        keys.add("1");
+                        keys.add(null);
+                        keys.add("3");
+                        return keys;
+                    }
+                });
 
-        MapConfig mapConfig = config.getMapConfig(name);
-        MapStoreConfig mapStoreConfig = new MapStoreConfig();
-        mapStoreConfig.setEnabled(true);
-        mapStoreConfig.setImplementation(new MapLoader() {
-            @Override
-            public Object load(Object key) {
-                if (key.equals("1")) {
-                    return "1";
-                }
-                if (key.equals("2")) {
-                    return "2";
-                }
-                if (key.equals("3")) {
-                    return "3";
-                }
-                return null;
-            }
-
-            @Override
-            public Map loadAll(Collection keys) {
-                Map val = new HashMap();
-                if (keys.contains("1")) {
-                    val.put("1", "1");
-                }
-                if (keys.contains("2")) {
-                    val.put("2", "2");
-                }
-                if (keys.contains("3")) {
-                    val.put("3", "3");
-                }
-                return val;
-            }
+        Config config = new Config();
+        config.getMapConfig(name)
+                .setMapStoreConfig(mapStoreConfig);
 
-            @Override
-            public Iterable loadAllKeys() {
-                List keys = new ArrayList();
-                keys.add("1");
-                keys.add(null);
-                keys.add("3");
-                return keys;
-            }
-        });
-        mapStoreConfig.setInitialLoadMode(MapStoreConfig.InitialLoadMode.LAZY);
-        mapConfig.setMapStoreConfig(mapStoreConfig);
         HazelcastInstance instance = createHazelcastInstance(config);
         IMap<String, String> map = instance.getMap(name);
 
-        expectedException.expect(NullPointerException.class);
-        expectedException.expectMessage(CoreMatchers.is("Key loaded by a MapLoader cannot be null."));
-        map.size();
+        try {
+            map.size();
+            fail("Expected a NPE due to a null key in a MapLoader");
+        } catch (NullPointerException e) {
+            assertEquals("Key loaded by a MapLoader cannot be null.", e.getMessage());
+        }
 
-        assertEquals(2, map.size());
+        assertEquals(0, map.size());
```

Comment:
@tombujok These asserts were not executed before, since the `ExpectedException` usage ends the tests immediately. After fixing this by doing a manual exception check, the asserts had to be fixed. But this looks so wrong. How can a map have to entries, but a `size()` of 0? @tombujok can you please have a look at this?

Reply:
The `null` key from `loadAllKeys()` results in the `NullPointerException` being thrown as the [particular _batch_ that contains the `null` key is sent for loading](https://github.com/vbekiaris/hazelcast/blob/16bd03c7794322b5989d1bd5641c80352c1117cb/hazelcast/src/main/java/com/hazelcast/map/impl/MapKeyLoader.java#L340-L343). Since the default load chunk size is 1000, in this test no keys are loaded on initial load. If you `setProperty(GroupProperty.MAP_LOAD_CHUNK_SIZE.getName(), "1")` on this test's `Config` you'll notice that the size of the map after initial loading is 1 (so key "1" was actually loaded and initial loading terminated with NPE when processing the second key which was `null`).